docs/summary.md

```diff
@@ -0,0 +1,203 @@
+## Apollo Link
+
+Apollo Link is an extensible standard interface for modifying control flow of GraphQL queries and fetching GraphQL results.
+
+## Basic Usage
+
+Links can be used as a stand-alone client or with most major GraphQL clients.
+
+### Stand-alone
+
+`execute` accepts a standard GraphQL request with a query string or AST.
+`execute` returns an [Observable](https://github.com/zenparsing/zen-observable#api) that can be subscribed to.
+Links use observables to support GraphQL subscriptions and live queries, in addition to single response queries and mutations.
+
+```js
+import {
+  execute,
+  HttpLink,
+} from 'apollo-link';
+
+const uri = 'http://api.githunt.com/graphql';
+const link = new HttpLink({ uri });
+
+execute(link, operation).subscribe({
+  next: data => console.log(`received data ${data}`),
+  error: error => console.log(`received error ${error}`),
+  complete: () => console.log(`complete`),
+})
+```
+
+`next` will receive GraphQL errors, while `error` be called on a network error.
+
+### Apollo Client
+
+Links can be used as a replacement for the current network interface.
+
+```js
+import {
+  execute,
+  HttpLink,
+} from 'apollo-link';
+
+const uri = 'http://api.githunt.com/graphql';
+const link = new HttpLink({uri});
+
+const client = new ApolloClient({
+  networkInterface: link,
+});
+```
+
+The `HttpLink` constructor accepts an optional fetch function and defaults to an `ApolloFetch` from [apollo-fetch](https://github.com/apollographql/apollo-fetch).
+Additionally, `ApolloFetch` provides a transition point for middleware and afterware.
+
+### GraphiQL
+
+GraphiQL provides a simple way of testing a link.
+
+```js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import '../node_modules/graphiql/graphiql.css'
+import GraphiQL from 'graphiql';
+
+import {
+  execute,
+} from 'apollo-link';
+
+const uri = 'http://api.githunt.com/graphql';
+const link = new HttpLink({uri});
+
+ReactDOM.render(
+  <GraphiQL fetcher={(operation) => execute(link, operation)}/>,
+  document.body,
+);
+```
+
+If using `create-react-app`, your `index.js` would look similar to above.
+
+## Adding Functionality
+
+Links can composed together to form a new link using `ApolloLink.from`.
+
+### Retry
+
+Attempts to resend a GraphQL request when failed and fails after a certain time
+
+```js
+import {
+  ApolloLink,
+  execute,
+  HttpLink,
+  RetryLink,
+} from 'apollo-link'
+
+const uri = 'http://api.githunt.com/graphql';
+
+const link = ApolloLink.from([
+  new RetryLink();
+  new HttpLink({ uri });
+])
+
+<GraphiQL fetcher={(operation) => execute(link, operation)}/>,
+
+const client = new ApolloClient({
+  networkInterface: link,
+});
+```
+
+### Authentication
+
+Adds authentication headers, using the context
```

Comment:
I think this is missing a bit of an explanation that `HttpLink` ends up reading the context to get the headers.